Doc/library/sqlite3.rst

```diff
@@ -222,14 +222,17 @@ Module functions
 
 .. function:: complete_statement(statement)
 
-   Returns ``True`` if the string *statement* contains one or more complete SQL
-   statements terminated by semicolons. It does not verify that the SQL is
-   syntactically correct, only that there are no unclosed string literals and the
-   statement is terminated by a semicolon.
+   Return ``True`` if the string *statement* appears to contain
+   one or more complete SQL statements.
+   No syntactic verification or parsing of any kind is performed,
+   other than checking that there are no unclosed string literals
+   and the statment is terminated by a semicolon.
 
-   This can be used to build a shell for SQLite, as in the following example:
+   This function may be useful during command-line input
+   to determine if the entered text seems to form a complete SQL statement,
+   or if additional input is needed before calling :meth:`~Cursor.execute`.
 
-   .. literalinclude:: ../includes/sqlite3/complete_statement.py
+   See :source:`Lib/sqlite3/__main__.py` for example use.
```

Comment:
The linked example doesn't seem particularly useful, since it's not too obvious
* what the input is
* why the function returns `True` when it's not complete

The fact that the function doesn't have a docstring doesn't help either.

---

Since `complete_statement` seems trivial enough (it doesn't need a DB or any complex setup before using it), what about just adding something like this (with whatever query you think would better illustrate "completeness"):
```py
>>> sqlite3.complete_statement('SELECT foo FROM bar;')
True
>>> sqlite3.complete_statement('SELECT foo')
False
```

Reply:
> The linked example doesn't seem particularly useful, since it's not too obvious
> * what the input is
> * why the function returns `True` when it's not complete
>
> The fact that the function doesn't have a docstring doesn't help either.

Good points; this can easily be improved by telling the user which `__main__.py` function to look for in the docs, and of course by adding docstrings to `__main__.py`.

A trivial example could work well for the 3.11 and 3.10 docs, which don't have the CLI.

Reply:
@ezio-melotti makes a great point; IMO, the minimal example would be good to include regardless, as it clearly and succinctly illustrates the purpose of the function, and when it is expected to be used. In the 3.12 docs, you could mention `__main__` (perhaps with a manual permalink to a specific line) as an additional, more complete example beyond that, but I certainly don't think it replaces the former in terms of function.

Reply:
> you could mention __main__ (perhaps with a manual permalink to a specific line)

FWIW ``:source:`Lib/sqlite3/__main__.py#L39` `` seems to work as a link, but it's not "perma" and the anchor is rendered too. 
``:source:`Lib/sqlite3/__main__.py <https://github.com/python/cpython/blob/3646f6cd880f8f91e189a2fe44a687798aa1fef1/Lib/sqlite3/__main__.py#L39>` `` appends the full link to `https://github.com/python/cpython/tree/main/...` so it doesn't work.  If you want to add a permalink, you'll have to use regular rst links (otherwise you can just use the minimal example).

Reply:
After pondering about this for a day, I'm not so sure it is a good idea to use the CLI as an example. I'm not totally opposed to it, though. Do we even need the trivial example? There are more complex functions in the sqlite3 API without examples. Perhaps a parameter listing would be clearer: we could then specify two `:returns:` directives.

Reply:
> Do we even need the trivial example?

Seeing now that CAM also suggest the trivial example. I'm fine with that; let's add it.

Reply:
I tried adding some explanatory comments to the CLI to make it better suited as an example. However, it is hard for me to see the comments actually serve their purpose, since I wrote the CLI in the first place; everything in there is obvious to me :)